### ISSUE TITLE
PROJ v7.0.0: wrong geoid file name for some vertical transformations from `PROJ` registry

### ISSUE THREAD

**Issue by and-marsh** [OSGeo/PROJ#2112]
#### Example of problem
Hello, it looks like some height transformation doesn't resolve geoid file name correctly. I have all new pack of `PROJ` data installed, but instead of using new name of geoid, for example `dk_sdfe_dvr90.tif`,  `PROJ` still tries to find `dvr90.gtx` in example below.

```c
./projinfo -s EPSG:5799 -t EPSG:4937 --spatial-test intersects

Candidate operations found: 1
-------------------------------------
Operation No. 1:

PROJ:EPSG_5799_TO_EPSG_4937, DVR90 height to ETRS89, unknown accuracy, Denmark - onshore, at least one grid missing

PROJ string:
+proj=vgridshift +grids=dvr90.gtx +multiplier=1
...

Grid dvr90.gtx needed but not found on the system. Can be obtained at https://cdn.proj.org/dk_sdfe_dvr90.tif

```

I found this issue for entries of `grid_transformation` with `auth_name` `PROJ`:
```
EPSG_5799_TO_EPSG_4937
EPSG_5733_TO_EPSG_4937
EPSG_5317_TO_EPSG_4937
EPSG_5613_TO_EPSG_4977
EPSG_8089_TO_EPSG_5323
EPSG_8089_TO_EPSG_4945
EPSG_8089_TO_EPSG_8085
```

#### Environment Information
 - PROJ version (v7.0.0)

**Comment by kbevers** [comment]
Renaming the grids is a feature not a bug, since we want to be able to be backwards compatible at the same time as being able to organize the grid files better. Bugs are of course possible in this part of the code... So, which package have you installed? `proj-data-1.0`? And where have you installed it? What is the content of `PROJ_LIB`.

**Comment by rouault** [comment]
I consider the current behaviour as a bug. Fix in PR #2113

**Comment by and-marsh** [comment]
@kbevers Hey, yes package is `proj-data-1.0` and it unpacked in original `PROJ` data directory. All other transformations with geoids works correctly except these i mentioned.
